front-end/src/app/shared/services/contact.service.ts

```diff
@@ -85,19 +85,24 @@ export class ContactService implements TableListService<Contact> {
       .pipe(map((response) => CommitteeLookupResponse.fromJSON(response)));
   }
 
-  public checkFecIdForUniqness(fec_id: string): Observable<boolean> {
-    return fec_id ? this.apiService.get<boolean>(`/contacts/fec_id_is_unique/${fec_id}/`) : of(true);
+  public checkFecIdForUniqness(fec_id: string, contact_id?: string): Observable<boolean> {
+    if (fec_id) {
+      const url = `/contacts/fec_id_is_unique/?fec_id=${fec_id}`.concat(
+        contact_id ? `&contact_id=${contact_id}` : '');
+      return this.apiService.get<boolean>(url);
+    }
+    return of(true);
   }
 
   public fecIdValidator: AsyncValidatorFn = (control: AbstractControl) => {
     return of(control.value).pipe(
-      debounceTime(500),
       switchMap((fecId) =>
-        this.checkFecIdForUniqness(fecId).pipe(
-          map((isUnique: boolean) => {
-            return isUnique ? null : { fecIdMustBeUnique: true };
-          })
-        )
+        this.checkFecIdForUniqness(fecId, (control.parent?.get(
+          'contact_1')?.value as Contact)?.id).pipe(
```

Comment:
is contact_1 available when editing a contact on the manage contacts page (not in a transaction form)?

Reply:
I'm a bit concerned that we are hardcoding 'contact_1' in the contact service and the contact form component which ideally should be agnostic as to which transaction contact record they are working with. Can the contact form work with additional contacts beyond the first one declared for a transaction?